File: ed2/gara3/monete/att/monete.c
#include <stdio.h>
#include <assert.h>

int raccogli(int N, int M, int monete[], int A[], int B[]) {
    // Mettete qui il codice della soluzione
    return 42;
}


#define MAXN 10000
#define MAXM 100000

int monete[MAXN];
int A[MAXM], B[MAXM];

int main() {
    FILE *fr, *fw;
    int N, M, i;

    fr = fopen("input.txt", "r");
    fw = fopen("output.txt", "w");
    assert(2 == fscanf(fr, "%d%d", &N, &M));
    for(i=0; i<N; i++)
        assert(1 == fscanf(fr, "%d", &monete[i]));
    for(i=0; i<M; i++)
        assert(2 == fscanf(fr, "%d%d", &A[i], &B[i]));

    fprintf(fw, "%d\n", raccogli(N, M, monete, A, B));
    fclose(fr);
    fclose(fw);
    return 0;
}

File: ed2/gara3/monete/att/monete.pas
function raccogli(N, M: longint; var monete, A, B: array of longint): longint;
begin
    (* Mettete qui il codice della soluzione *)
    raccogli := 42;
end;


const
    MAXN = 10000;
    MAXM = 100000;
var
    N, M, i : longint;
    monete  : array[0..MAXN-1] of longint;
    A, B    : array[0..MAXM-1] of longint;
    fr, fw  : text;

begin
    assign(fr, 'input.txt');
    assign(fw, 'output.txt');
    reset(fr);
    rewrite(fw);
    readln(fr, N, M);
    for i:=0 to N-1 do
        read(fr, monete[i]);
    readln;
    for i:=0 to M-1 do
        readln(fr, A[i], B[i]);

    writeln(fw, raccogli(N, M, monete, A, B));
    close(fr);
    close(fw);
end.
